### PR TITLE
MSFT contributions March 2020

### PR DESCRIPTION
**Pull request by jankrassnigg**

### CODE REVIEW

File: Code/Engine/Foundation/Types/Implementation/Variant_inl.h [hunk 1]
```diff
@@ -6,166 +6,199 @@ EZ_ALWAYS_INLINE ezVariant::ezVariant()
 }
 
 EZ_ALWAYS_INLINE ezVariant::ezVariant(const ezVariant& other)
+  : ezVariant()
```
Comment (C-Core) [inline Code/Engine/Foundation/Types/Implementation/Variant_inl.h]: I don't understand this change. Seems like a complete waste to first default init all values and then overwrite them again with the proper values.

File: Code/UnitTests/FoundationTest/Strings/FormatStringTest.cpp [hunk 2]
```diff
@@ -131,8 +151,8 @@ EZ_CREATE_SIMPLE_TEST(Strings, FormatString)
     CompareSnprintf(perfLog, ezFmt("{0}", ezArgF(23.123456789, 2)), "%.2f", 23.123456789);
     CompareSnprintf(perfLog, ezFmt("{0}", ezArgI(123456789, 20, true)), "%020i", 123456789);
     CompareSnprintf(perfLog, ezFmt("{0}", ezArgI(123456789, 20, true, 16)), "%020X", 123456789);
-    CompareSnprintf(perfLog, ezFmt("{0}", ezArgU(1234567890987ll, 30, false, 16)), "% 30llx", 1234567890987ll);
-    CompareSnprintf(perfLog, ezFmt("{0}", ezArgU(1234567890987ll, 30, false, 16, true)), "% 30llX", 1234567890987ll);
+    CompareSnprintf(perfLog, ezFmt("{0}", ezArgU(1234567890987ll, 30, false, 16)), "% 30x", 1234567890987ll);
```
[the remark this answers is not in the document]
Comment (maapp) [reply Code/UnitTests/FoundationTest/Strings/FormatStringTest.cpp]: Yep, that one was for a Semmle detected problem.

File: Code/Engine/Foundation/Types/Implementation/Variant_inl.h (hunk 1 above)
Comment (jankrassnigg) [reply Code/Engine/Foundation/Types/Implementation/Variant_inl.h]: Reverting this for now, as it has no benefit other than making some static analyzer happy. Need to follow up on why the analyzer complained about it.

File: Code/UnitTests/FoundationTest/Strings/FormatStringTest.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jankrassnigg) [reply Code/UnitTests/FoundationTest/Strings/FormatStringTest.cpp]: fixed

**Pull request merged**